radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokenclasses/TokenState.java

```diff
@@ -39,10 +48,33 @@ public BigDecimal getTotalSupply() {
 		return totalSupply;
 	}
 
+	public TokenSupplyType getTokenSupplyType() {
+		return tokenSupplyType;
+	}
+
 	public BigDecimal getMaxSupply() {
 		return totalSupply;
 	}
 
+	@Override
+	public int hashCode() {
+		return Objects.hash(name, iso, description, tokenSupplyType, totalSupply);
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (!(o instanceof TokenState)) {
+			return false;
+		}
+
+		TokenState tokenState = (TokenState) o;
+		return this.name.equals(tokenState.name)
```

Comment:
Better as a general rule would be to use `Objects.equals(a, b)` which will avoid NPE's for `null` fields.

Reply:
right!